pkg/sdk/pipeline.go

```diff
@@ -46,6 +46,13 @@ func NewClientOptions(connection Connection) *arm.ClientOptions {
 				&removeAuthorizationHeaderPolicy{},
 			},
 			Transport: connection.Client(),
+			// When updating azcore to 1.11.1 from 1.7.0, we saw that HTTPS check for Authentication was added.
+			// Link to the check: https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azcore/runtime/policy_bearer_token.go#L118
+			//
+			// This check was failing for some unit tests because the ARM requests are made over HTTP and the bearer token is being sent in the header.
+			// This is a temporary fix to allow sending the bearer token over HTTP.
```

Comment:
Do we have a longer term fix in mind? If not, then I'm not sure if the comment about "temporary fix" is conveying the intended message.